File: Tatsuki_Yabe/Q4/Swift/solution.swift
class Solution {
    func moveZeroes(_ nums: inout [Int]) {
        var i = 0
        nums.forEach {
            if ($0 != 0) {
                nums[i] = $0
                i+=1
            }
        }

        for k in i ..< nums.count {
            nums[k] = 0
        }
    }
}

File: Tatsuki_Yabe/Q4/Go/solution.go
package main

func moveZeroes(nums []int) {
	left := 0
	for _, v := range nums {
		if v != 0 {
			nums[left] = v
			left += 1
		}
	}
	for k := left; k < len(nums); k++ {
		nums[k] = 0
	}
}
